### PR TITLE
feat(openshift-etcd-backup): add podSecurityContext support

### PR DESCRIPTION
**Pull request by vmaillot**
# Description

Add support of `podSecurityContext` for additionnalGroups support as described [here](https://docs.openshift.com/container-platform/4.7/storage/persistent_storage/persistent-storage-nfs.html#storage-persistent-storage-nfs-group-ids_persistent-storage-nfs).

# Issues

# Checklist

* [x] I updated the version in Chart.yaml
* [x] I updated applicable README.md files using  `pre-commit run -a`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] If I updated a dependency tool, or app, this PR contains a short summary of the changes I'm pulling
* [x] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

### CODE REVIEW

File: charts/openshift-etcd-backup/templates/cronjob.yaml [hunk 1]
```diff
@@ -11,6 +11,8 @@ spec:
       backoffLimit: 0
       template:
         spec:
+          securityContext:
+            supplementalGroups: {{ .Values.securityContext.supplementalGroups }}
```
Comment (hairmare) [inline charts/openshift-etcd-backup/templates/cronjob.yaml:15]:
```suggestion
          securityContext:
            {{- toYaml .Values.podSecurityContext | nindent 12 }}
```

File: charts/openshift-etcd-backup/values.yaml [hunk 2]
```diff
@@ -65,6 +65,8 @@ securityContext:
   privileged: true
   # -- Set user ID
   runAsUser: 0
+  # -- Define supplemental groups
+  supplementalGroups: [""]
```
Comment (hairmare) [inline charts/openshift-etcd-backup/values.yaml:69]:
```suggestion

podSecurityContext:
  # -- Define supplemental groups
  supplementalGroups: [""]
```
do we need an empty string as default here or would an empty array (or even an empty `podSecurityContext`) make more sense?

**Review by hairmare — CHANGES_REQUESTED** [review]

File: charts/openshift-etcd-backup/values.yaml (hunk 2 above)
Comment (vmaillot) [reply charts/openshift-etcd-backup/values.yaml:69]: you're right, an ampty `podSecurityContext` would fit best!

File: charts/openshift-etcd-backup/templates/cronjob.yaml [hunk 3]
```diff
@@ -11,6 +11,8 @@ spec:
       backoffLimit: 0
       template:
         spec:
+          securityContext:
+             {{- toYaml .Values.podSecurityContext | nindent 12 }}
```
Comment (tongpu) [inline charts/openshift-etcd-backup/templates/cronjob.yaml:15]: One space too much here:
```suggestion
            {{- toYaml .Values.podSecurityContext | nindent 12 }}
```

File: charts/openshift-etcd-backup/values.yaml [hunk 4]
```diff
@@ -66,6 +66,10 @@ securityContext:
   # -- Set user ID
   runAsUser: 0
 
+podSecurityContext:
```
Comment (tongpu) [inline charts/openshift-etcd-backup/values.yaml:69]:
```suggestion
# -- Configure SecurityContext of the pod started by the job
podSecurityContext: {}
```

**Review by tongpu — CHANGES_REQUESTED** [review]

File: charts/openshift-etcd-backup/templates/cronjob.yaml (hunk 3 above)
Comment (vmaillot) [reply charts/openshift-etcd-backup/templates/cronjob.yaml:15]: wow, nice catch, thanks!

**Review by hairmare — APPROVED** [review]

**Review by tongpu — APPROVED** [review]

**Pull request merged**